java-checks/src/main/java/org/sonar/java/checks/IndentationAfterConditionalCheck.java

```diff
@@ -0,0 +1,172 @@
+import org.sonar.plugins.java.api.tree.StatementTree;
+import org.sonar.plugins.java.api.tree.SyntaxTrivia;
+import org.sonar.plugins.java.api.tree.Tree;
+import org.sonar.plugins.java.api.tree.WhileStatementTree;
+
+@Rule(key = "S3973")
+public class IndentationAfterConditionalCheck extends BaseTreeVisitor implements JavaFileScanner {
+
+  private JavaFileScannerContext context;
+
+  @Override
+  public void scanFile(JavaFileScannerContext context) {
+    this.context = context;
+    scan(context.getTree());
+  }
+
+  @Override
+  public void visitIfStatement(IfStatementTree tree) {
+    if (thereIsNoIssue(tree)) {
+      return;
+    }
+    if (checkElseStatementReports(tree)) {
+      context.reportIssue(this, tree.elseKeyword(),
+        "Use curly braces or indentation to denote the code conditionally executed by this \"" + tree.elseKeyword().text() + "\".");
+    }
+    report(tree.ifKeyword(), tree.closeParenToken(), tree.ifKeyword().text(), findSecondariesToReport(tree));
+    super.visitIfStatement(tree);
+  }
+
+  @Override
+  public void visitWhileStatement(WhileStatementTree tree) {
+    if (thereIsNoIssue(tree)) {
+      return;
+    }
+    report(tree.whileKeyword(), tree.closeParenToken(), tree.whileKeyword().text(), findSecondariesToReport(tree));
+    super.visitWhileStatement(tree);
+  }
+
+  @Override
+  public void visitForStatement(ForStatementTree tree) {
+    if (thereIsNoIssue(tree)) {
+      return;
+    }
+    report(tree.forKeyword(), tree.closeParenToken(), tree.forKeyword().text(), findSecondariesToReport(tree));
+    super.visitForStatement(tree);
+  }
+
+  @Override
+  public void visitForEachStatement(ForEachStatement tree) {
+    if (thereIsNoIssue(tree)) {
+      return;
+    }
+    report(tree.forKeyword(), tree.closeParenToken(), tree.forKeyword().text(), findSecondariesToReport(tree));
+    super.visitForEachStatement(tree);
+  }
+
+  private static boolean thereIsNoIssue(Tree tree) {
+    return (blockExists(tree) || checkIdentation(tree));
+  }
+
+  public void report(Tree firstTree, Tree secondTree, String operationName, List<JavaFileScannerContext.Location> flow) {
+    context.reportIssue(this, firstTree, secondTree,
+      "Use curly braces or indentation to denote the code conditionally executed by this \"" + operationName + "\".", flow, null);
+  }
+
+  private static boolean checkElseStatementReports(IfStatementTree tree) {
+    if (tree.elseKeyword() != null && !tree.elseStatement().is(Tree.Kind.IF_STATEMENT)) {
+      return tree.elseKeyword().column() >= tree.elseStatement().firstToken().column();
+    }
+    return false;
+  }
+
+  private static List<JavaFileScannerContext.Location> findSecondariesToReport(Tree tree) {
+    Tree parentTree = tree.parent();
+    List<Tree> secondaryLinesToReport = new ArrayList<>();
+    StatementTree statement = returnStatement(tree);
+    secondaryLinesToReport.add(statement);
+    if (parentTree.is(Tree.Kind.BLOCK)) {
```

Comment:
What about the case where the parent is an if/for/while/switch-case ?
A more general solution could be to implement a `BaseTreeVisitor` to scan the parent tree and check until what line the following statements/expressions go to